File: LoginDemo/LoginDemo/ViewModel/Login/LoginValidate.vb
Imports System.ComponentModel.DataAnnotations

Public Class NotEmptyCheck
    Inherits ValidationAttribute
    Public Overrides Function IsValid(value As Object) As Boolean
        Dim name = TryCast(value, String)
        If String.IsNullOrEmpty(name) Then
            Return False
        End If
        Return True
    End Function

    Public Overrides Function FormatErrorMessage(name As String) As String
        Return "不能为空"
    End Function
End Class

Public Class UserNameExists
    Inherits ValidationAttribute
    Public Overrides Function IsValid(value As Object) As Boolean
        Dim name = TryCast(value, String)
        If Not IsNothing(name) AndAlso name.Contains("qige") Then
            Return True
        End If
        Return False
    End Function

    Public Overrides Function FormatErrorMessage(name As String) As String
        Return "用户名必须包含qige"
    End Function
End Class

File: LoginDemo/LoginDemo/ViewModel/Login/LoginValidate.cs
using System.ComponentModel.DataAnnotations;

namespace LoginDemo.ViewModel.Login
{
    public class NotEmptyCheck : ValidationAttribute
    {
        public override bool IsValid(object value)
        {
            var name = value as string;
            if (string.IsNullOrEmpty(name))
            {
                return false;
            }
            return true;
        }

        public override string FormatErrorMessage(string name)
        {
            return "不能为空";
        }
    }

    public class UserNameExists : ValidationAttribute
    {
        public override bool IsValid(object value)
        {
            var name = value as string;
            if (name.Contains("abc"))
            {
                return true;
            }
            return false;
        }

        public override string FormatErrorMessage(string name)
        {
            return "用户名必须包含abc";
        }
    }
}
